### ISSUE TITLE
Bulk upload of tag deployments not working for project 78

### ISSUE THREAD

**Issue by eageldart** [MotusDev/Motus-TO-DO#1107]
I am trying to use the bulk metadata upload tool to add metadata to project 78. Once I complete my upload, I get the following message:

**Upload Complete**
Your data has been successfully transferred to the Motus Database.
**New deployments created::** 0
**Existing deployments that were updated::** 0

When I check the tags I tried to change in the Tag Management Page, my changes are not made.

I tried making very small changes (e.g., added a band number) and it still did not update this existing deployment. I believe the researchers on this project are having the same problem since they told me they had updated all of their 2021 deployments, but they are still missing metadata. I have attached the Motus Tags Template that I am working from (downloaded from motus). 
[MotusTagsTemplate-78.csv](https://github.com/MotusDev/Motus-TO-DO/files/8069713/MotusTagsTemplate-78.csv)

I have been using the bulk upload tool to work on several other projects and have not had this problem.

**Comment by denislepage** [comment]
Here's the error that I am getting when trying to run the import (which is failing silently):

"Start date is greater than or equal to the end date"

This currently applies to 2 tags, both of which are flagged as test tags. Although there is a validation in place for this condition, it is being ignored because these 2 are test tags.

This is the problematic section (last condition where test != 1).

```
SELECT a.rowNum,'Start/End Dates' as field_name,tsStart,tsEnd,
dbo.ts2datetime(tsStart) as startDT,dbo.ts2datetime(tsEnd) as endDT, tagDeployID,tagID, confirm_dt_start 
INTO #temp_ts
FROM api_tag_deployments a 
                LEFT JOIN #temp_error b ON a.apiBatchID=b.apiBatchID AND a.rowNum=b.rowNum AND b.field_name like 'utc%'
WHERE b.error_code IS NULL and a.apiBatchID = @apiBatchID AND test != 1 
```

The reason it is failing silently, which is a separate issue, is because the proc currently catches the error and rollbacks to update/insert, but doesn't return an error to the calling method, which leaves the values in their original state (0 updated, 0 inserted). 

I tried to add a RAISERROR statement in the catch block, but it's being handled silently somewhere else in Java, it seems. 

Anyway.

1. Immediate fix is to set the end date later than the start date/time in the file, even just by 1 minute. I just tried the project 78 file with success after setting utcMinuteEnd = 1

2. Second fix would be to either add this condition to generate an api_error record in the validation, or allow tags to have the same start and end date in the deployment table.

3. I don't have an immediate solution to propose, but we should ideally prevent this proc from failing silently, so I least we can catch any other conditions we missed during validation.

**Comment by eageldart** [comment]
Thank you for looking into this, Denis. 

I tried immediate fix #1 and it worked. I will keep an eye on this for future silent errors in the meantime. 

Would you like me to close this issue, or keep it open to continue to discuss immediate fixes #2 and #3?

**Comment by leberrigan** [comment]
Seeing as you're immediate issue has been resolved, I will close this issue in deference to #1127 and #1128.
